Kickstarter-iOS/Views/UIView+ShimmerLoading.swift

```diff
@@ -0,0 +1,174 @@
+import Foundation
+import Library
+import ObjectiveC
+import UIKit
+
+private let gradientLayerName = "ksr_shimmer_gradientLayer"
+
+private enum ShimmerConstants {
+  enum Locations {
+    static let start: [NSNumber] = [-1.0, -0.5, 0.0]
+    static let end: [NSNumber] = [1.0, 1.5, 2.0]
+  }
+
+  enum Animation {
+    static let movingAnimationDuration: CFTimeInterval = 1.25
+    static let delayBetweenAnimationLoops: CFTimeInterval = 0.3
+  }
+}
+
+private struct AssociatedKeys {
+  static var isLoading = "isLoading"
+  static var shimmersWhenLoading = "shimmersWhenLoading"
+  static var shimmerLayers = "shimmerLayers"
+}
+
+protocol ShimmerLoading: AnyObject {
+  func startLoading()
+  func stopLoading()
+}
+
+extension UIView {
+  var shimmersWhenLoading: Bool {
```

Comment:
Would like to think of a way to not do this to _all_ `UIView`s.

Reply:
Actually a simpler way of doing this is probably just to have the conforming view return an array of views to add the shimmer to. I’ll make that change.

Reply:
Agree this would be nicer 👍

Reply:
Also the fewer associated objects the better 😅